packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/backup.service.js

```diff
@@ -0,0 +1,166 @@
+import find from 'lodash/find';
+import get from 'lodash/get';
+import map from 'lodash/map';
+import remove from 'lodash/remove';
+import sortBy from 'lodash/sortBy';
+
+import { BACKUP_OFFER_LEGACY, CATALOG_INFO } from './backup.constants';
+import Backup from './backup.class';
+import BackupOffer from './backup-offer.class';
+
+export default class BackupService {
+  /* @ngInject */
+  constructor($q, OvhApiDedicatedCloudDatacenter, OvhApiOrder, WucOrderCartService) {
+    this.$q = $q;
+    this.backupApi = OvhApiDedicatedCloudDatacenter.Backup().v6();
+    this.catalogApi = OvhApiOrder.CatalogFormatted().v6();
+    this.WucOrderCartService = WucOrderCartService;
+  }
+
+  addConfigToCart(cartItem, datacenterId, offerType) {
+    return this.$q
+      .all([
+        this.WucOrderCartService.addConfigurationItem(
+          cartItem.cartId,
+          cartItem.itemId,
+          'datacenter_id',
+          datacenterId,
+        ).$promise,
+        this.WucOrderCartService.addConfigurationItem(
+          cartItem.cartId,
+          cartItem.itemId,
+          'offer_type',
+          offerType,
+        ).$promise,
+      ])
+      .then(() => cartItem);
+  }
+
+  assignCart(cart) {
+    return this.WucOrderCartService.assignCart(cart.cartId);
+  }
+
+  checkoutCart(cart) {
+    return this.WucOrderCartService.checkoutCart(cart.cartId, {
+      autoPayWithPreferredPaymentMethod: true,
+    });
+  }
+
+  createCart(ovhSubsidiary) {
+    return this.WucOrderCartService.createNewCart(ovhSubsidiary);
+  }
+
+  getOrderableBackupOption(productId) {
+    return this.WucOrderCartService.getProductServiceOptions(
+      'privateCloud',
+      productId,
+    ).$promise.then((addons) =>
+      find(addons, { family: CATALOG_INFO.ADDON_FAMILY_NAME }),
+    );
+  }
+
+  addOptionToCart(option, cart, productId) {
+    return this.WucOrderCartService.addProductServiceOptionToCart(
+      cart.cartId,
+      'privateCloud',
+      productId,
+      {
+        cartId: cart.cartId,
+        duration: get(option, 'prices[0].duration'),
+        planCode: option.planCode,
+        pricingMode: get(option, 'prices[0].pricingMode'),
```

Comment:
Please use the `capacity` property to get prices, we already had the case where some prices were added and so the first one wasn't the one to take anymore

Reply:
This is done. Thanks for the input